parser/parser_expression_test.go

```diff
@@ -254,6 +254,53 @@ func TestFuncCallMissingComma(t *testing.T) {
 	assertErrorAt(t, p, 0, ", expected, but got 2")
 }
 
+func TestFuncCallOnMember(t *testing.T) {
+	e := parseExpressionFromInput(t, "a.f()")
+	assertFuncCall(t, e, "a.f")
+}
+
+// Element Access Expression
+// -------------------------
+
+func TestElementAccess(t *testing.T) {
+	e := parseExpressionFromInput(t, "arr[0]")
+	assertElementAccess(t, e, "arr", "0")
+}
+
+func TestElementAccessWithDesignator(t *testing.T) {
+	e := parseExpressionFromInput(t, "arr[pos]")
+	assertElementAccess(t, e, "arr", "pos")
+}
+
+func TestElementAccessWithExpression(t *testing.T) {
+	e := parseExpressionFromInput(t, "arr[1 + 2]")
+	assertElementAccess(t, e, "arr", "(1 + 2)")
+}
+
+func TestElementAccessOnMember(t *testing.T) {
+	e := parseExpressionFromInput(t, "a.arr[pos]")
+	assertElementAccess(t, e, "a.arr", "pos")
+}
+
+// Member Access Expression
+// ------------------------
+
+func TestMemberAccess(t *testing.T) {
+	e := parseExpressionFromInput(t, "a.x")
+	assertMemberAccess(t, e, "a", "x")
+}
+
+func TestMultipleMemberAccess(t *testing.T) {
+	e := parseExpressionFromInput(t, "a.x.y")
+	assertMemberAccess(t, e, "a.x", "y")
+}
+
+func TestInvalidMemberAccess(t *testing.T) {
+	p := newParserFromInput("a.0")
+	p.parseExpression()
+	assertHasError(t, p)
```

Comment:
Use `assertErrorAt` for checking errors

Reply:
Fixed https://github.com/bazo-blockchain/lazo/pull/55/commits/50bbce4be94f797c8c6332a68f58e75a84466e61